samples/SocialSample/Startup.cs

```diff
@@ -227,48 +230,16 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
 
                         var user = JObject.Parse(await response.Content.ReadAsStringAsync());
 
-                        var identifier = user.Value<string>("id");
-                        if (!string.IsNullOrEmpty(identifier))
-                        {
-                            context.Identity.AddClaim(new Claim(
-                                ClaimTypes.NameIdentifier, identifier,
-                                ClaimValueTypes.String, context.Options.ClaimsIssuer));
-                        }
-
-                        var userName = user.Value<string>("login");
-                        if (!string.IsNullOrEmpty(userName))
-                        {
-                            context.Identity.AddClaim(new Claim(
-                                ClaimsIdentity.DefaultNameClaimType, userName,
-                                ClaimValueTypes.String, context.Options.ClaimsIssuer));
-                        }
-
-                        var name = user.Value<string>("name");
-                        if (!string.IsNullOrEmpty(name))
-                        {
-                            context.Identity.AddClaim(new Claim(
-                                "urn:github:name", name,
-                                ClaimValueTypes.String, context.Options.ClaimsIssuer));
-                        }
-
-                        var email = user.Value<string>("email");
-                        if (!string.IsNullOrEmpty(email))
-                        {
-                            context.Identity.AddClaim(new Claim(
-                                ClaimTypes.Email, email,
-                                ClaimValueTypes.Email, context.Options.ClaimsIssuer));
-                        }
-
-                        var link = user.Value<string>("url");
-                        if (!string.IsNullOrEmpty(link))
-                        {
-                            context.Identity.AddClaim(new Claim(
-                                "urn:github:url", link,
-                                ClaimValueTypes.String, context.Options.ClaimsIssuer));
-                        }
+                        context.MapClaims(user);
                     }
                 }
-            });
+            };
+            githubOptions.ClaimMaps.MapJsonKey(ClaimTypes.NameIdentifier, "id");
```

Comment:
Would we lose any meaning if we called the property Claims since all of the methods are called Map?  Just looks a bit odd for it to be ClaimsMaps.Map everywhere?